### PR TITLE
refactored event creation and processing using a workflow approach

### PR DESCRIPTION
**Pull request by mekilis**

### CODE REVIEW

File: internal/pkg/metrics/metrics.go [hunk 1]
```diff
@@ -45,9 +44,6 @@ func RequestDuration() *prometheus.HistogramVec {
 }
 
 func RegisterQueueMetrics(q queue.Queuer, db database.Database) {
-	Reg().MustRegister(
-		metrics.NewQueueMetricsCollector(q.(*redisqueue.RedisQueue).Inspector()),
-	)
```
Comment (jirevwe) [inline internal/pkg/metrics/metrics.go:50]: Why was this removed?

**Review by jirevwe — APPROVED** [review]

File: worker/task/process_event_channel.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+		err = eventQueue.Write(convoy.MatchEventSubscriptionsProcessor, convoy.EventWorkflowQueue, job)
+		if err != nil {
+			log.FromContext(ctx).WithError(err).Errorf("[asynq]: an error occurred while matching event subs")
+		}
+
+		return err
+	}
+}
+
+func MatchSubscriptionsAndCreateEventDeliveries(channels map[string]EventChannel, endpointRepo datastore.EndpointRepository, eventRepo datastore.EventRepository, projectRepo datastore.ProjectRepository, eventDeliveryRepo datastore.EventDeliveryRepository, eventQueue queue.Queuer, subRepo datastore.SubscriptionRepository, deviceRepo datastore.DeviceRepository, licenser license.Licenser) func(context.Context, *asynq.Task) error {
```
Comment (danvixent) [inline worker/task/process_event_channel.go:107]: All these plenty params, can we create a struct to hold them? This goes for all the other methods with a lot of params as well.

Comment (mekilis) [reply worker/task/process_event_channel.go:107]: Noted.
